torch/nn/utils/stateless.py

```diff
@@ -72,12 +101,17 @@ def _remove_swap(module, name: str, full_path: str) -> None:
 def _reparametrize_module(
     module: 'torch.nn.Module',
     parameters_and_buffers: Dict[str, Tensor],
+    tie_weights: bool,
 ) -> Iterator[None]:
-    orig_tensors_to_replacements: Dict[Tensor, Tensor] = {}
+    tied_weights_map = _create_tied_weights_map(module, parameters_and_buffers.keys()) if tie_weights else {}
     for name, tensor in parameters_and_buffers.items():
         _apply_func_submodules(
-            _create_swap_params(parameters_and_buffers, orig_tensors_to_replacements),
+            _create_swap_params(parameters_and_buffers),
             module, name.split("."), name, (tensor,))
+    for tied_name, user_given_name in tied_weights_map.items():
+        _apply_func_submodules(
+            _create_swap_params(parameters_and_buffers),
+            module, tied_name.split("."), user_given_name, (torch.empty(()),))
```

Comment:
Why are we creating an empty tensor here?

Reply:
To match the signature of `_create_swap_params` (which requires a Tensor) but let me just change the signature to be `Optional[Tensor]` so we don't have to pass anything